#Actividad 1.2
-------------------------------------

####Angel Rubén Vázquez Rivera
######A01735407

###Importar librerias

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

###Cargar archivo

In [2]:
df = pd.read_excel("Datos de Facturación.xlsx")
df

CVE_DOC    CVE_CLPV STATUS  CVE_VEND               FECHAELAB  FECHA_DOC  \
0      F29414        2465      C       1.0 2019-09-03 08:58:42.000 2019-09-03   
1      F29415        3926      C       1.0 2019-09-03 09:55:14.000 2019-09-03   
2      F29416        3926      C       1.0 2019-09-03 13:25:33.000 2019-09-03   
3      F29417        3926      C       1.0 2019-09-03 13:33:23.000 2019-09-03   
4      F29418        6813      C       2.0 2019-09-03 14:23:45.000 2019-09-03   
...       ...         ...    ...       ...                     ...        ...   
10890  F40306        8842      E       2.0 2023-03-06 16:09:39.013 2023-03-06   
10891  F40307        6218    NaN       2.0 2023-03-06 16:38:42.933 2023-03-06   
10892     NaN        5926      E       2.0 2023-03-06 17:02:33.390 2023-03-06   
10893  F40309         361      E       2.0 2023-03-07 12:43:49.567 2023-03-07   
10894  F40310        3011      E       2.0 2023-03-07 12:45:58.407 2023-03-07   

                 FECHA_ENT  FECHA_VEN FECHA_CANCELA  CAN_TOT  DES_TOT  DES_FIN  
0      2019-09-03 00:00:00 2019-09-18    2019-09-04  3980.00      0.0      0.0  
1      2019-09-03 00:00:00 2019-10-03    2019-09-03  1550.00      0.0      0.0  
2      2019-09-03 00:00:00 2019-10-03    2019-09-03  1550.00      0.0      0.0  
3      2019-09-03 00:00:00 2019-10-03    2019-09-17  1550.00      0.0      0.0  
4      2019-09-03 00:00:00 2019-09-03    2019-09-10  2435.44      0.0      0.0  
...                    ...        ...           ...      ...      ...      ...  
10890  2023-03-01 00:00:00 2023-03-06           NaT  3272.00      0.0      0.0  
10891  2023-03-06 00:00:00 2023-03-06           NaT   504.15      0.0      0.0  
10892  2023-03-06 00:00:00 2023-03-06           NaT   275.68      0.0      0.0  
10893  2023-03-07 00:00:00 2023-03-07           NaT   129.30      0.0      0.0  
10894  2023-03-01 00:00:00 2023-03-07           NaT   517.20      0.0      0.0  

[10895 rows x 12 columns]

###Filtro 1
CVE_CLPV (Valores entre 1000 y 2000)

En este caso primero se reparan algunos datos faltantes o se modifican otros para poder ser manipulados de manera adecuada.

Despues con el filtro se toman todos los registros de la columna CVE_CLPV que tengan valores entre 1000 y 2000

In [3]:
# Limpieza de datos con replace
df.CVE_CLPV = df.CVE_CLPV.replace({'O':'0'}, regex=False)
df.CVE_CLPV = df.CVE_CLPV.replace({'6781|':'6781'}, regex=False)
df.CVE_CLPV = df.CVE_CLPV.replace({'3612C':'3612'}, regex=False)
df.CVE_CLPV = df.CVE_CLPV.replace({'      ':'0'}, regex=False)
df.CVE_CLPV = df.CVE_CLPV.replace({'2385A':'2385'}, regex=False)
df.CVE_CLPV = df.CVE_CLPV.replace({'2510A':'2510'}, regex=False)
df.CVE_CLPV = df.CVE_CLPV.replace({'8440Q':'8440'}, regex=False)
df.CVE_CLPV = df.CVE_CLPV.replace({'M':'0'}, regex=False)

filtro1 = df[(df["CVE_CLPV"].astype(int) >= 1000) & (df["CVE_CLPV"].astype(int) <= 2000)]
display(filtro1)


CVE_DOC    CVE_CLPV STATUS  CVE_VEND               FECHAELAB  FECHA_DOC  \
18     F29432        1965      C       2.0 2019-09-05 11:05:04.000 2019-09-05   
22     F29436        1965      C       2.0 2019-09-06 12:01:36.000 2019-09-06   
65     F29479        1240      E       2.0 2019-09-11 18:38:45.000 2019-09-11   
71     F29485        1965      C       2.0 2019-09-12 12:14:28.000 2019-09-12   
72     F29486        1965      E       2.0 2019-09-12 12:25:52.000 2019-09-12   
...       ...         ...    ...       ...                     ...        ...   
10821  F40237        1757      E       2.0 2023-02-24 15:18:17.570 2023-02-24   
10839  F40255        1583      E       2.0 2023-02-28 09:15:45.753 2023-02-28   
10845  F40261        1786      E       2.0 2023-02-28 11:15:32.527 2023-02-28   
10871  F40287        1453      E       1.0 2023-03-01 12:58:23.397 2023-03-01   
10876  F40292        1082      E       6.0 2023-03-02 11:00:54.770 2023-03-02   

                 FECHA_ENT  FECHA_VEN FECHA_CANCELA   CAN_TOT  DES_TOT  \
18     2019-09-05 00:00:00 2019-09-20    2019-09-06    976.50     0.00   
22     2019-09-06 00:00:00 2019-09-21    2019-09-12    976.50     0.00   
65     2019-09-11 00:00:00 2019-09-11           NaT   4135.00     0.00   
71     2019-09-12 00:00:00 2019-09-27    2019-09-12    976.50   146.48   
72     2019-09-12 00:00:00 2019-09-27           NaT    976.50   146.48   
...                    ...        ...           ...       ...      ...   
10821  2023-02-13 00:00:00 2023-02-24           NaT  10088.40     0.00   
10839  2023-02-28 00:00:00 2023-03-30           NaT   3467.35     0.00   
10845  2023-03-03 00:00:00 2023-02-28           NaT   4663.80     0.00   
10871  2023-02-27 00:00:00 2023-03-31           NaT   9000.00     0.00   
10876  2023-03-01 00:00:00 2023-03-02           NaT   3262.00     0.00   

       DES_FIN  
18         0.0  
22         0.0  
65         0.0  
71         0.0  
72         0.0  
...        ...  
10821      0.0  
10839      0.0  
10845      0.0  
10871      0.0  
10876      0.0  

[581 rows x 12 columns]

###Guardar en CSV


In [4]:
filtro1.to_csv("Filtro1.csv")

###Filtro 2
CVE_VEND

Filtra todos los registros donde CVE_VEND no tanga como clave 4 o 5

In [5]:
filtro2 = df[(df["CVE_VEND"] != "4") | (df["CVE_VEND"] != "5")]
filtro2

CVE_DOC    CVE_CLPV STATUS  CVE_VEND               FECHAELAB  FECHA_DOC  \
0      F29414        2465      C       1.0 2019-09-03 08:58:42.000 2019-09-03   
1      F29415        3926      C       1.0 2019-09-03 09:55:14.000 2019-09-03   
2      F29416        3926      C       1.0 2019-09-03 13:25:33.000 2019-09-03   
3      F29417        3926      C       1.0 2019-09-03 13:33:23.000 2019-09-03   
4      F29418        6813      C       2.0 2019-09-03 14:23:45.000 2019-09-03   
...       ...         ...    ...       ...                     ...        ...   
10890  F40306        8842      E       2.0 2023-03-06 16:09:39.013 2023-03-06   
10891  F40307        6218    NaN       2.0 2023-03-06 16:38:42.933 2023-03-06   
10892     NaN        5926      E       2.0 2023-03-06 17:02:33.390 2023-03-06   
10893  F40309         361      E       2.0 2023-03-07 12:43:49.567 2023-03-07   
10894  F40310        3011      E       2.0 2023-03-07 12:45:58.407 2023-03-07   

                 FECHA_ENT  FECHA_VEN FECHA_CANCELA  CAN_TOT  DES_TOT  DES_FIN  
0      2019-09-03 00:00:00 2019-09-18    2019-09-04  3980.00      0.0      0.0  
1      2019-09-03 00:00:00 2019-10-03    2019-09-03  1550.00      0.0      0.0  
2      2019-09-03 00:00:00 2019-10-03    2019-09-03  1550.00      0.0      0.0  
3      2019-09-03 00:00:00 2019-10-03    2019-09-17  1550.00      0.0      0.0  
4      2019-09-03 00:00:00 2019-09-03    2019-09-10  2435.44      0.0      0.0  
...                    ...        ...           ...      ...      ...      ...  
10890  2023-03-01 00:00:00 2023-03-06           NaT  3272.00      0.0      0.0  
10891  2023-03-06 00:00:00 2023-03-06           NaT   504.15      0.0      0.0  
10892  2023-03-06 00:00:00 2023-03-06           NaT   275.68      0.0      0.0  
10893  2023-03-07 00:00:00 2023-03-07           NaT   129.30      0.0      0.0  
10894  2023-03-01 00:00:00 2023-03-07           NaT   517.20      0.0      0.0  

[10895 rows x 12 columns]

###Guardar en CSV


In [6]:
filtro2.to_csv("Filtro2.csv")

###Filtro 3
FECHA_ENT (fechas de 13/04/2022 12:00:00)

Filtro todos los registros donde en la columna FECHA_ENT tenga como valor 2022-02-28 00:00:00

In [8]:
filtro3 = df["FECHA_ENT"] == pd.to_datetime("2022-02-28 00:00:00")
resultados = df[filtro3]

# Mostrar el DataFrame resultante
display(resultados)


CVE_DOC    CVE_CLPV STATUS  CVE_VEND               FECHAELAB  FECHA_DOC  \
8077  F37493        4598      E       1.0 2022-02-28 10:56:02.460 2022-02-28   
8078  F37494        1943      C       2.0 2022-02-28 11:34:00.367 2022-02-28   
8081  F37497        1943      C       2.0 2022-02-28 12:05:42.377 2022-02-28   
8083  F37499        8573      E       2.0 2022-02-28 12:25:29.833 2022-02-28   
8084  F37500        8249      E       2.0 2022-02-28 12:32:46.687 2022-02-28   
8090  F37506        1757      E       2.0 2022-02-28 13:23:45.440 2022-02-28   
8091  F37507        3108      E       2.0 2022-02-28 13:52:43.383 2022-02-28   
8092  F37508        6232      E       2.0 2022-02-28 14:46:56.940 2022-02-28   
8095  F37511        8593      E       2.0 2022-02-28 15:20:58.040 2022-02-28   
8096  F37512        4766      E       2.0 2022-02-28 15:54:16.277 2022-02-28   
8100  F37516        6179      E       3.0 2022-02-28 16:33:54.413 2022-02-28   
8103  F37519        5150      E       3.0 2022-02-28 17:09:39.213 2022-02-28   
8104  F37520        4527      E       2.0 2022-02-28 17:14:57.487 2022-02-28   
8107  F37523        5926      E       2.0 2022-02-28 17:36:54.363 2022-02-28   
8133  F37549        3016      E       1.0 2022-03-04 15:08:49.747 2022-03-04   
8148  F37564        4695      E       1.0 2022-03-08 09:15:20.507 2022-03-08   
8204  F37620        8316      E       3.0 2022-03-14 17:00:43.247 2022-03-14   

                FECHA_ENT  FECHA_VEN FECHA_CANCELA     CAN_TOT  DES_TOT  \
8077  2022-02-28 00:00:00 2022-03-30           NaT  44259.0000      0.0   
8078  2022-02-28 00:00:00 2022-02-28    2022-02-28     89.6800      0.0   
8081  2022-02-28 00:00:00 2022-02-28    2022-02-28     89.6800      0.0   
8083  2022-02-28 00:00:00 2022-02-28           NaT    997.5400      0.0   
8084  2022-02-28 00:00:00 2022-02-28           NaT   2508.7000      0.0   
8090  2022-02-28 00:00:00 2022-02-28           NaT   2586.2100      0.0   
8091  2022-02-28 00:00:00 2022-02-28           NaT    675.4000      0.0   
8092  2022-02-28 00:00:00 2022-02-28           NaT    941.3965      0.0   
8095  2022-02-28 00:00:00 2022-02-28           NaT    218.9700      0.0   
8096  2022-02-28 00:00:00 2022-02-28           NaT   9450.0000      0.0   
8100  2022-02-28 00:00:00 2022-02-28           NaT   5951.7000      0.0   
8103  2022-02-28 00:00:00 2022-02-28           NaT   2040.0000      0.0   
8104  2022-02-28 00:00:00 2022-02-28           NaT   4315.0000      0.0   
8107  2022-02-28 00:00:00 2022-02-28           NaT    451.6100      0.0   
8133  2022-02-28 00:00:00 2022-03-04           NaT   1083.7200      0.0   
8148  2022-02-28 00:00:00 2022-05-07           NaT   5515.0000      0.0   
8204  2022-02-28 00:00:00 2022-03-14           NaT  24570.3000      0.0   

      DES_FIN  
8077      0.0  
8078      0.0  
8081      0.0  
8083      0.0  
8084      0.0  
8090      0.0  
8091      0.0  
8092      0.0  
8095      0.0  
8096      0.0  
8100      0.0  
8103      0.0  
8104      0.0  
8107      0.0  
8133      0.0  
8148      0.0  
8204      0.0

###Guardar en CSV

In [9]:
filtro3.to_csv("Filtro3.csv")

###Filtro 4
CAN_TOT (cantidades menores a 5951.7) o estatus E

Filtro que me permite obtener todos los registros donde en la columna CAN_TOT los valores sean inferiores a 59517 o su valor en la columna STATUS sea igual a E

In [11]:
filtro4 = df[(df["CAN_TOT"].astype(float) < 59517) | (df["STATUS"] == "E")]
filtro4

CVE_DOC    CVE_CLPV STATUS  CVE_VEND               FECHAELAB  FECHA_DOC  \
0      F29414        2465      C       1.0 2019-09-03 08:58:42.000 2019-09-03   
1      F29415        3926      C       1.0 2019-09-03 09:55:14.000 2019-09-03   
2      F29416        3926      C       1.0 2019-09-03 13:25:33.000 2019-09-03   
3      F29417        3926      C       1.0 2019-09-03 13:33:23.000 2019-09-03   
4      F29418        6813      C       2.0 2019-09-03 14:23:45.000 2019-09-03   
...       ...         ...    ...       ...                     ...        ...   
10890  F40306        8842      E       2.0 2023-03-06 16:09:39.013 2023-03-06   
10891  F40307        6218    NaN       2.0 2023-03-06 16:38:42.933 2023-03-06   
10892     NaN        5926      E       2.0 2023-03-06 17:02:33.390 2023-03-06   
10893  F40309         361      E       2.0 2023-03-07 12:43:49.567 2023-03-07   
10894  F40310        3011      E       2.0 2023-03-07 12:45:58.407 2023-03-07   

                 FECHA_ENT  FECHA_VEN FECHA_CANCELA  CAN_TOT  DES_TOT  DES_FIN  
0      2019-09-03 00:00:00 2019-09-18    2019-09-04  3980.00      0.0      0.0  
1      2019-09-03 00:00:00 2019-10-03    2019-09-03  1550.00      0.0      0.0  
2      2019-09-03 00:00:00 2019-10-03    2019-09-03  1550.00      0.0      0.0  
3      2019-09-03 00:00:00 2019-10-03    2019-09-17  1550.00      0.0      0.0  
4      2019-09-03 00:00:00 2019-09-03    2019-09-10  2435.44      0.0      0.0  
...                    ...        ...           ...      ...      ...      ...  
10890  2023-03-01 00:00:00 2023-03-06           NaT  3272.00      0.0      0.0  
10891  2023-03-06 00:00:00 2023-03-06           NaT   504.15      0.0      0.0  
10892  2023-03-06 00:00:00 2023-03-06           NaT   275.68      0.0      0.0  
10893  2023-03-07 00:00:00 2023-03-07           NaT   129.30      0.0      0.0  
10894  2023-03-01 00:00:00 2023-03-07           NaT   517.20      0.0      0.0  

[10886 rows x 12 columns]

###Guardar en CSV

In [12]:
filtro4.to_csv("Filtro4.csv")

###Filtro 5
Solo las columnas CVE_DOC, FECHA_ENT, FECHA_VEN y CAN_TOT

Se filtran los registros absolutos pero solo de 4 columnas especificas

In [13]:
filtro5 = df.loc[:,["CVE_DOC", "FECHA_ENT", "FECHA_VEN", "CAN_TOT"]]
filtro5

CVE_DOC            FECHA_ENT  FECHA_VEN  CAN_TOT
0      F29414  2019-09-03 00:00:00 2019-09-18  3980.00
1      F29415  2019-09-03 00:00:00 2019-10-03  1550.00
2      F29416  2019-09-03 00:00:00 2019-10-03  1550.00
3      F29417  2019-09-03 00:00:00 2019-10-03  1550.00
4      F29418  2019-09-03 00:00:00 2019-09-03  2435.44
...       ...                  ...        ...      ...
10890  F40306  2023-03-01 00:00:00 2023-03-06  3272.00
10891  F40307  2023-03-06 00:00:00 2023-03-06   504.15
10892     NaN  2023-03-06 00:00:00 2023-03-06   275.68
10893  F40309  2023-03-07 00:00:00 2023-03-07   129.30
10894  F40310  2023-03-01 00:00:00 2023-03-07   517.20

[10895 rows x 4 columns]

###Guardar en CSV

In [14]:
filtro5.to_csv("Filtro5.csv")

###Filtro 6
Solo las filas 7001 a 7099

Se filtran los registros entre 7001 y 7100 de todas las columnas

In [15]:
filtro6 = df.iloc[7001:7100,:]
filtro6

CVE_DOC    CVE_CLPV STATUS  CVE_VEND               FECHAELAB  FECHA_DOC  \
7001  F36417        5271      E       3.0 2021-10-15 08:40:29.183 2021-10-15   
7002  F36418        5746      E       2.0 2021-10-15 08:51:03.463 2021-10-15   
7003  F36419        2163      E       3.0 2021-10-15 08:56:23.280 2021-10-15   
7004  F36420        1429      E       1.0 2021-10-15 09:01:08.393 2021-10-15   
7005  F36421        5039      E       1.0 2021-10-15 09:07:07.103 2021-10-15   
...      ...         ...    ...       ...                     ...        ...   
7095  F36511        4480      E       2.0 2021-10-27 13:07:43.223 2021-10-27   
7096  F36512        8186      E       2.0 2021-10-27 13:22:23.980 2021-10-27   
7097  F36513        4403      E       2.0 2021-10-27 13:35:40.677 2021-10-27   
7098  F36514        5589      E       2.0 2021-10-27 14:02:54.630 2021-10-27   
7099  F36515        5589      E       2.0 2021-10-27 14:04:26.797 2021-10-27   

                FECHA_ENT  FECHA_VEN FECHA_CANCELA   CAN_TOT  DES_TOT  DES_FIN  
7001  2021-10-06 00:00:00 2021-11-14           NaT   8555.24      0.0      0.0  
7002  2021-10-14 00:00:00 2021-10-15           NaT    620.77      0.0      0.0  
7003  2021-10-15 00:00:00 2021-10-15           NaT    560.00      0.0      0.0  
7004  2021-10-15 00:00:00 2021-12-14           NaT  23625.00      0.0      0.0  
7005  2021-10-08 00:00:00 2022-01-13           NaT   3308.78      0.0      0.0  
...                   ...        ...           ...       ...      ...      ...  
7095  2021-10-27 00:00:00 2021-11-11           NaT   1137.50      0.0      0.0  
7096  2021-10-27 00:00:00 2021-10-27           NaT     94.84      0.0      0.0  
7097  2021-10-27 00:00:00 2021-10-27           NaT    847.00      0.0      0.0  
7098  2021-10-27 00:00:00 2021-10-27           NaT    965.00      0.0      0.0  
7099  2021-10-26 00:00:00 2021-10-27           NaT   2106.00      0.0      0.0  

[99 rows x 12 columns]

###Guardar en CSV

In [16]:
filtro6.to_csv("Filtro6.csv")

###Filtro 7
Index= CVE_VEND(valores 1,2) (COlumna: FECHALAB)

Se filtran todos los registros de la columna FECHAELAB donde la columna con el indice 1.0 o 2.0

In [21]:
df1 = pd.read_excel("Datos de Facturación.xlsx", index_col="CVE_VEND")
filtro7 = df1.loc[[1.0, 2.0], ["FECHAELAB"]]
filtro7



FECHAELAB
CVE_VEND                        
1.0      2019-09-03 08:58:42.000
1.0      2019-09-03 09:55:14.000
1.0      2019-09-03 13:25:33.000
1.0      2019-09-03 13:33:23.000
1.0      2019-09-04 09:15:12.000
...                          ...
2.0      2023-03-06 16:09:39.013
2.0      2023-03-06 16:38:42.933
2.0      2023-03-06 17:02:33.390
2.0      2023-03-07 12:43:49.567
2.0      2023-03-07 12:45:58.407

[8665 rows x 1 columns]

###Guardar en CSV

In [22]:
filtro7.to_csv("Filtro7.csv")